php/class-fm-bylines-cli.php

```diff
@@ -36,9 +51,188 @@ public function migrate_coauthors( $args, $assoc_args ) {
+						print_r( $args );
+						$byline_post = get_posts( $args );
+					}
+
+					// Create a byline.
+					if ( empty( $byline_post[0]->ID ) ) {
+						// Generate the name of the byline.
+						if ( ! empty( $cap_meta['cap-display_name'][0] ) ) {
+							$byline_display_name = $cap_meta['cap-display_name'][0];
+						} elseif ( ! empty( $mapped_user->name ) ) {
+							$byline_display_name = $mapped_user->name;
+						} else {
+							$byline_display_name = $cap_term->slug;
+						}
+
+						// Create a byline post.
+						$byline_args = array(
+							'post_title' => $byline_display_name,
+							'post_name' => $byline_slug,
+							'post_type' => FM_Bylines()->name,
+							'post_status' => 'publish',
+						);
+						$byline_id = wp_insert_post( $byline_args );
+						if ( ! empty( $byline_id ) ) {
+							// Set byline metadata to CAP metadata.
+							// You can use the filter hook fm_bylines_cli_cap_meta to add in the missing meta fields such as twitter, linkedin & shortbio if they exist in the CAP metadata.
+							update_post_meta( $byline_id, 'fm_bylines_names', array(
+								'first_name' => ! empty( $cap_meta['cap-first_name'][0] ) ? $cap_meta['cap-first_name'][0] : '',
+								'last_name' => ! empty( $cap_meta['cap-first_name'][0] ) ? $cap_meta['cap-last_name'][0] : '',
+							) );
+
+							update_post_meta( $byline_id, 'fm_bylines_contact_info', array(
+								'email' => ! empty( $cap_meta['cap-user_email'][0] ) ? $cap_meta['cap-user_email'][0] : '',
+								'website' => ! empty( $cap_meta['cap-website'][0] ) ? $cap_meta['cap-website'][0] : '',
+								'twitter' => ! empty( $cap_meta['cap-twitter'][0] ) ? $cap_meta['cap-twitter'][0] : '',
+								'linkedin' => ! empty( $cap_meta['cap-linkedin'][0] ) ? $cap_meta['cap-linkedin'][0] : '',
+							) );
+
+							update_post_meta( $byline_id, 'fm_bylines_about', array(
+								'bio' => ! empty( $cap_meta['cap-description'][0] ) ? $cap_meta['cap-description'][0] : '',
+								'short-bio' => ! empty( $cap_meta['cap-shortbio'][0] ) ? $cap_meta['cap-shortbio'][0] : '',
+							) );
+
+							if ( ! empty( $mapped_user ) ) {
+								update_post_meta( $byline_id, 'fm_bylines_user_mapping', $mapped_user->ID );
+							}
+						} else {
+							WP_CLI::line( "Falied to insert byline: {$byline_slug}" );
+						}
+					} else {
+						$byline_id = $byline_post[0]->ID;
+					}
+
+					if ( ! empty( $cap_post[0]->ID ) ) {
+						do_action( 'fm_bylines_cli_post_cap_migration', $byline_id, $cap_meta, $cap_term, $cap_post[0] );
+					} else {
+						do_action( 'fm_bylines_cli_post_cap_migration', $byline_id, $cap_meta, $cap_term, false );
+					}
+
+					// Clean legacy data. Don't delete by default. This will migrate the featured image as well.
+					if ( $clear_data ) {
+						// We are just flipping the post id here.
+						global $wpdb;
+						$wpdb->update( $wpdb->postmeta, array( 'post_id' => $byline_id ), array( 'post_id' => $cap_post[0]->ID ), array( '%d' ), array( '%d' ) );
```

Comment:
will im still thinking we should loop this instead of flip them. ur thoughts? @willgladstone